pypesto/visualize/parameters.py

```diff
@@ -381,14 +381,13 @@ def handle_inputs(
         res.get(INNER_PARAMETERS, None) for res in result.optimize_result.list
     ]
     if any(inner_xs):
-        # search for first non-empty inner_xs to obtain inner_xs_names
-        for inner_xs_idx in inner_xs:
-            if inner_xs_idx is not None:
-                inner_xs_names = list(inner_xs_idx.keys())
-                break
-        # fill inner_xs with nan if no inner_xs are available
+        inner_xs_names = next(
+            list(inner_xs_idx.keys())
+            for inner_xs_idx in inner_xs
+            if inner_xs_idx is not None
+        )
         inner_xs = [
-            np.full(len(inner_xs_names), np.nan)
+            [np.nan for i in range(len(inner_xs_names))]
```

Comment:
Can't remember honestly. Probably is equivalent.

Reply:
Is this equivalent? In the empty edge case 391 will be a numpy array (empty) and 390 will be a list (empty). If this does not matter, then it's the same I would say.

Reply:
I believe it does not matter. `len(inner_xs_names)` will never be 0. It will not come into this if statement if that's the case. Then the `inner_xs` will just be None and it will pass nicely.